src/main/kotlin/graphics/scenery/controls/behaviours/VRTouch.kt

```diff
@@ -121,14 +102,47 @@ open class VRTouch(
                                 controller.children.first(),
                                 device,
                                 { scene.discover(scene, { n -> n.getAttributeOrNull(Touchable::class.java) != null }) },
-                                if (vibrate) fun(){ (hmd as? OpenVRHMD)?.vibrate(device) } else fun(){})
+                                if (vibrate) fun(){ (hmd as? OpenVRHMD)?.vibrate(device); onTouch?.invoke() } else onTouch)
                             future.complete(touchBehaviour)
                         }
                     }
                 }
             }
             return future
         }
+
+        /**
+         * Apply the [Touchable.onHoldChangeDiffuseTo] color.
+         * If you are calling this manually make sure [unapplySelectionColor] will be called later.
+         */
+        fun applySelectionColor(node: Node) {
```

Comment:
If calling this manually could lead to problems, maybe make it private? Or is there a better solution, such as handing over a lambda, with applySelectionColor() being called, then the lambda, then restore?

Reply:
There are scenarios where you might want to call those functions externally. The unapply function I am actually using already in my code.